labman/db/process.py

```diff
@@ -3293,14 +3336,37 @@ def generate_prep_information(self):
                     result[nk] = val
 
                 # format some final fields
-                result['platform'] = equipment[
-                    result.pop('platform_id')]['description']
+                result['platform'] = 'Illumina'
+                result['instrument_model'] = ''
+                if assay == self._amplicon_assay_type:
+                    result['extraction_robot'] = '%s_%s' % (
+                        result.pop('epmotion_robot'),
+                        result.pop('kingfisher_robot'))
+                    result['well_description'] = '%s_%s_%s' % (
+                        result['sample_plate'], sid, result['well_id'])
+                    result['primer_plate'] = result[
+                        'primer_plate'].split(' ')[-1]
+                    mgps = marker_gene_primer_set[result.pop('primer_set_id')]
+                    result['PRIMER'] = '%s%s' % (
+                        mgps['linker_sequence'], mgps['fwd_primer_sequence'])
+                    result['pcr_primers'] = 'FWD:%s; REV:%s' % (
+                        mgps['fwd_primer_sequence'],
+                        mgps['rev_primer_sequence'])
+                    result['linker'] = mgps['linker_sequence']
+                    result['target_gene'] = mgps['target_gene']
+                    result['target_subfragment'] = mgps['target_subfragment']
+                    result['library_construction_protocol'] = (
+                        'Illumina EMP protocol {0} amplification of {1}'
+                        ' {2}'.format(mgps['region'], mgps['target_gene'],
```

Comment:
What is the region here? target_gene and target_subfragment make sense, however I don't believe the EMP protocol allows for variation in the subfragment of a given gene. 

Is it possible that the fwd/rev primers set on like 3352 will contradict the library construction protocol?

Reply:
The field called "region" in the database holds the region of the gene that is amplified by the actual, particular primer pair in use; its values were provided by Gail and are:
16S V4: "515fbc, 806r"
18S V9: "1391f/EukBr"
ITS 1/2: "ITS1fbc, ITS2r"

The only way I can think of that the primer sets could contradict the library construction protocol was if we linked the primer records in the db to the wrong marker gene primer set; this is the one-time set-up done in db_patch_manual.sql and populate_prod_db.sql, so hopefully we do it right once and we're golden :)

Reply:
Ah okay, so the wording is "Illumina EMP protocol 515fbc, 806r amplication of 16S V4" or to that effect. 

And thanks for the clarification on the lack of a possible contradiction :)